src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.cs

```diff
@@ -367,9 +367,22 @@ private bool ShouldVisitNamespace(ISymbol containingSymbol)
 
         private bool IncludeNamedType(INamedTypeSymbol namedType)
         {
-            return
-                namedType != null &&
-                (!namedType.IsScriptClass || format.CompilerInternalOptions.IncludesOption(SymbolDisplayCompilerInternalOptions.IncludeScriptType));
+            if (namedType is null)
+            {
+                return false;
+            }
+
+            if (namedType.IsScriptClass && !format.CompilerInternalOptions.IncludesOption(SymbolDisplayCompilerInternalOptions.IncludeScriptType))
+            {
+                return false;
+            }
+
+            if (namedType == semanticModelOpt?.Compilation.ScriptGlobalsType)
```

Comment:
>namedType == semanticModelOpt?.Compilation.ScriptGlobalsType [](start = 16, length = 60)

This looks like an incomplete solution. Can't we tell if the type is a ScriptGlobalsType from the type itself?

Reply:
Why incomplete? I don't think we can. It's just a regular PE type.

Reply:
> Why incomplete? I don't think we can. It's just a regular PE type.

Incomplete in a sense that we can omit the type name only when we are given SemanticModel. I wasn't clear to me that that was the request. It sounded more general to me.

I also have another question. Is it valid to omit the type anywhere in the compilation or only in a specific portion of code within it?

---
In reply to: [424634984](https://github.com/dotnet/roslyn/pull/43425#discussion_r424634984) [](ancestors = 424634984)

Reply:
> Incomplete in a sense that we can omit the type name only when we are given SemanticModel. It wasn't clear to me that that was the request. It sounded more general to me.

I see. SemanticModel is provided in the scenarios we are trying to address - that is the various places where the IDE displays the minimal type name.

> I also have another question. Is it valid to omit the type anywhere in the compilation or only in a specific portion of code within it?

It's similar to the implicit script class that's handled a few lines above. We never display the synthesized script class (the `SymbolDisplayCompilerInternalOptions.IncludeScriptType` option is only used in tests). In the common case the script code does not have access to the script globals type symbol itself (it can't refer to it). It only has access to its members.

It is theoretically possible for the script to add a reference to the assembly that contains the script globals type and then access the type itself. In practice, I don't see why would anyone want to do that. If they do set it up this way, they'll not see the full name in quick info, completion, etc. I don't think that's a problem. The script globals type is designated by the host to be special, so it is.